bundles/org.eclipse.cdt.lsp.editor.ui.test/src/org/eclipse/cdt/lsp/editor/ui/test/clangd/ClangdConfigurationManagerTest.java

```diff
@@ -0,0 +1,185 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Bachmann electronic GmbH and others.
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ * Gesa Hentschke (Bachmann electronic GmbH) - initial implementation
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp.editor.ui.test.clangd;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.util.Arrays;
+
+import org.eclipse.cdt.lsp.editor.ui.clangd.ClangdConfigurationManager;
+import org.eclipse.cdt.lsp.editor.ui.test.TestUtils;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.io.TempDir;
+import org.yaml.snakeyaml.scanner.ScannerException;
+
+class ClangdConfigurationManagerTest {
+	private static final String RELATIVE_DIR_PATH_BUILD_DEFAULT = "build/default";
+	private static final String RELATIVE_DIR_PATH_BUILD_DEBUG = "build/debug";
+	private static final String EXPANDED_CDB_SETTING = "CompileFlags: {Add: -ferror-limit=500, CompilationDatabase: %s, Compiler: g++}\nDiagnostics:\n  ClangTidy: {Add: modernize*, Remove: modernize-use-trailing-return-type}\n";
+	private static final String DEFAULT_CDB_SETTING = "CompileFlags: {CompilationDatabase: %s}";
+	private static final String MODIFIED_DEFAULT_CDB_SETTING = DEFAULT_CDB_SETTING + "\n";
+	private static final String INVALID_YAML_SYNTAX_CONTAINS_TAB = "CompileFlags:\n\tCompilationDatabase: %s";
```

Comment:
The .clangd in CDT I added recently had tabs before the comments. clangd seems fine with it, but I have fixed it in CDT in https://github.com/eclipse-cdt/cdt/pull/382

TBH - not sure how tab got in there in the first place, but I suspect the default text editor in Eclipse with default settings. Maybe we should add `.clangd` to the set of yml files, or contribute our own language mapping

Reply:
> TBH - not sure how tab got in there in the first place, but I suspect the default text editor in Eclipse with default settings. Maybe we should add .clangd to the set of yml files, or contribute our own language mappin

Yepp, that's s a good point